yt_dlp/extractor/rai.py

```diff
@@ -735,3 +699,67 @@ def _real_extract(self, url):
         info.update(relinker_info)
 
         return info
+
+
+class RaiNewsIE(RaiIE):
+    _VALID_URL = r'https?://(www\.)?rainews\.it/.+?-(?P<id>%s)(?:-.+?)?\.html' % RaiBaseIE._UUID_RE
+    _TESTS = [{
+        # new rainews player (#3911)
+        'url': 'https://www.rainews.it/rubriche/24mm/video/2022/05/24mm-del-29052022-12cf645d-1ffd-4220-b27c-07c226dbdecf.html',
+        'info_dict': {
+            'id': '12cf645d-1ffd-4220-b27c-07c226dbdecf',
+            'ext': 'mp4',
+            'title': 'Puntata del 29/05/2022',
+            'duration': 1589,
+            'upload_date': '20220529',
+            'uploader': 'rainews',
+        }
+    }, {
+        # old content with fallback method to extract media urls
+        'url': 'https://www.rainews.it/dl/rainews/media/Weekend-al-cinema-da-Hollywood-arriva-il-thriller-di-Tate-Taylor-La-ragazza-del-treno-1632c009-c843-4836-bb65-80c33084a64b.html',
+        'info_dict': {
+            'id': '1632c009-c843-4836-bb65-80c33084a64b',
+            'ext': 'mp4',
+            'title': 'Weekend al cinema, da Hollywood arriva il thriller di Tate Taylor "La ragazza del treno"',
+            'description': 'I film in uscita questa settimana.',
+            'thumbnail': r're:^https?://.*\.png$',
+            'duration': 833,
+            'upload_date': '20161103'
+        },
+        'expected_warnings': ['unable to extract player_data'],
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+
+        player_data = self._search_json(
+            r'<rainews-player\s*data=\'', webpage, 'player_data', video_id,
+            transform_source=clean_html, fatal=False)
+        track_info = player_data.get('track_info')
+        relinker_url = player_data.get('mediapolis') or player_data.get('content_url') or None
+
+        if not relinker_url:
+            # fallback on old implementation for some old content
+            try:
+                return self._extract_from_content_id(video_id, url)
+            except GeoRestrictedError:
+                raise
+            except ExtractorError:
+                raise ExtractorError('Relinker URL not found')
+
+        relinker_info = self._extract_relinker_info(
+            urljoin(url, relinker_url), video_id)
+        self._sort_formats(relinker_info['formats'])
+
+        info = {
+            'id': video_id,
+            'title': track_info.get('title') or self._og_search_title(webpage),
+            'upload_date': unified_strdate(track_info.get('date')),
+            'uploader': strip_or_none(track_info.get('editor') or None),
+        }
+
+        info.update(relinker_info)
+
+        return info
```

Comment:
```suggestion
        return {
            'id': video_id,
            'title': track_info.get('title') or self._og_search_title(webpage),
            'upload_date': unified_strdate(track_info.get('date')),
            'uploader': strip_or_none(track_info.get('editor') or None),
            **relinker_info
        }
```

Reply:
should I update this solution also in the other extractor class, RaiIE, from which I copied this one?

Reply:
I don't want to bother you with modernizing old code, but go ahead if you like to

Reply:
ok, done this and the f-string too.